indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -1024,6 +1024,24 @@ List<Task> getTasks()
     }
   }
 
+  /**
+   * Returns the list of currently active tasks for the given datasource.
+   */
+  public List<Task> getActiveTasksForDatasource(String datasource)
+  {
+    giant.lock();
```

Comment:
The operations are simple and thus enough, so it's okay (and also necessary) to hold the giant lock.